src/09-word-problems.js

```diff
@@ -127,8 +178,57 @@ function compareLocations(address1, address2) {}
     //>   },
     //> ];
  */
-function gradeAssignments(assignments) {}
+//score rec / scoremax * 100
+// function gradeAssignments(assignments) {
+//   //wrap in nested if or try loop
+//   //add if includes PASSFAIL
+//   if (assignments.kind === "PASS-FAIL") {
+//     if (assignments.score.received === assignments.score.max) {
+//       //add a `status` of PASSED if the `score.received` equals the `score.max (3 ms)
+//       assignments.status = "PASSED";
+//     } else {
+//       assignments.status = "FAILED";
+//     }
+//   }
+//   //if the `score` is equal to or greater than 80% and include the percentage
+//   //round PERCENTAGE to one decimal integer
+//   else if (assignments.kind === "PERCENTAGE") {
+//     if (assignments.score.received >= assignments.score.max * 0.8) {
+//       assignments.status = `"PASSED: ${assignments}"`; //might need place holder
+//     } else {
+//       assignments.status = "FAILED: "; //might need place holder
+//     }
+//   }
+//   // for any other assignment, add a `status` that shows the `score.received` out of the `score.max`
+//   //can i make this the else?
+//   else {
+//     assignments.status =
+//       "SCORE: " + assignments.score.received / assignments.score.max;
+//   }
 
+//   return assignments;
+// }
+function gradeAssignments(assignments) {
+  for (let assign of assignments) {
+    if (assign.kind === "PASS-FAIL") {
```

Comment:
A compound boolean like this:

```javascript
(assign.kind === 'PASS-FAIL' && assign.score.received === assign.score.max)
```

is usually a lot more readable than these ifs inside each other ("nested" ifs).